feature-date/src/main/java/com/anytypeio/anytype/feature_date/viewmodel/DateViewModel.kt

```diff
@@ -0,0 +1,865 @@
+        viewModelScope.launch {
+            _dateId
+                .filterNotNull()
+                .collect { id ->
+                    val params = GetObject.Params(
+                        target = id,
+                        space = vmParams.spaceId
+                    )
+                    Timber.d("Start GetObject with params: $params")
+                    getObject.async(params).fold(
+                        onSuccess = { obj ->
+                            Timber.d("GetObject Success, obj:[$obj]")
+                            val timestampInSeconds =
+                                obj.details[id]?.getSingleValue<Double>(
+                                    Relations.TIMESTAMP
+                                )?.toLong()
+                            if (timestampInSeconds != null) {
+                                _dateTimestamp.value = timestampInSeconds
+                                val (formattedDate, _) = dateProvider.formatTimestampToDateAndTime(
+                                    timestamp = timestampInSeconds * 1000,
+                                )
+                                uiCalendarIconState.value = UiCalendarIconState.Visible(
+                                    timestampInSeconds = TimestampInSeconds(timestampInSeconds)
+                                )
+                                uiHeaderState.value = UiHeaderState.Content(
+                                    title = formattedDate
+                                )
+                            }
+                        },
+                        onFailure = { e -> Timber.e(e, "GetObject Error") }
+                    )
+                }
+        }
+    }
+
+    private fun proceedWithGettingDateByTimestamp(timestamp: Long, action: (Struct?) -> Unit) {
+        val params = ObjectDateByTimestamp.Params(
+            space = vmParams.spaceId,
+            timestamp = timestamp
+        )
+        Timber.d("Start ObjectDateByTimestamp with params: [$params]")
+        viewModelScope.launch {
+            objectDateByTimestamp.async(params).fold(
+                onSuccess = { dateObject ->
+                    Timber.d("ObjectDateByTimestamp Success, dateObject: [$dateObject]")
+                    action(dateObject)
+                },
+                onFailure = { e -> Timber.e(e, "ObjectDateByTimestamp Error") }
+            )
+        }
+    }
+    //endregion
+
+    //region Subscription
+    private fun subscriptionId() = "date_object_subscription_${vmParams.spaceId}"
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private fun setupUiStateFlow() {
+        viewModelScope.launch {
+            combine(
+                _dateId.filterNotNull(),
+                _dateTimestamp.filterNotNull(),
+                _activeField.filterNotNull(),
+                restartSubscription
+            ) { dateId, timestamp, activeField, _ ->
+                loadData(dateId = dateId, timestamp = timestamp, field = activeField)
```

Comment:
we could map at this point `dateId, timestamp, activeField, _ ->` into search params. then load data when next operator is applied: `.flatMapLatest { searchParams -> loadData(searchParams) }`

Reply:
fixed